src/Aspire.Hosting/DistributedApplicationBuilder.cs

```diff
@@ -125,35 +105,62 @@ public DistributedApplicationBuilder(DistributedApplicationOptions options)
-        _innerBuilder.Services.AddLifecycleHook<DashboardManifestExclusionHook>();
+        if (ExecutionContext.IsRunMode)
+        {
+            // Dashboard
+            if (!options.DisableDashboard)
+            {
+                if (!IsDashboardUnsecured(_innerBuilder.Configuration))
+                {
+                    // Set a random API key for the OTLP exporter.
+                    // Passed to apps as a standard OTEL attribute to include in OTLP requests and the dashboard to validate.
+                    _innerBuilder.Configuration.AddInMemoryCollection(
+                        new Dictionary<string, string?>
+                        {
+                            ["AppHost:OtlpApiKey"] = TokenGenerator.GenerateToken()
+                        }
+                    );
+
+                    if (_innerBuilder.Configuration[KnownConfigNames.DashboardFrontendBrowserToken] is not { Length: > 0 } browserToken)
+                    {
+                        browserToken = TokenGenerator.GenerateToken();
+                    }
+
+                    // Set a random API key for the OTLP exporter.
+                    // Passed to apps as a standard OTEL attribute to include in OTLP requests and the dashboard to validate.
+                    _innerBuilder.Configuration.AddInMemoryCollection(
+                        new Dictionary<string, string?>
+                        {
+                            ["AppHost:BrowserToken"] = browserToken
+                        }
+                    );
+                }
 
-        // DCP stuff
-        _innerBuilder.Services.AddSingleton<ApplicationExecutor>();
-        _innerBuilder.Services.AddSingleton<IDashboardEndpointProvider, HostDashboardEndpointProvider>();
-        _innerBuilder.Services.AddSingleton<IDcpDependencyCheckService, DcpDependencyCheck>();
-        _innerBuilder.Services.AddHostedService<DcpHostService>();
+                _innerBuilder.Services.AddOptions<TransportOptions>().ValidateOnStart().PostConfigure(MapTransportOptionsFromCustomKeys);
+                _innerBuilder.Services.TryAddEnumerable(ServiceDescriptor.Singleton<IValidateOptions<TransportOptions>, TransportOptionsValidator>());
+                _innerBuilder.Services.AddSingleton<DashboardServiceHost>();
+                _innerBuilder.Services.AddHostedService(sp => sp.GetRequiredService<DashboardServiceHost>());
+                _innerBuilder.Services.AddLifecycleHook<DashboardLifecycleHook>();
+            }
 
-        // We need a unique path per application instance
-        _innerBuilder.Services.AddSingleton(new Locations());
-        _innerBuilder.Services.AddSingleton<IKubernetesService, KubernetesService>();
+            // DCP stuff
+            _innerBuilder.Services.AddSingleton<ApplicationExecutor>();
+            _innerBuilder.Services.AddSingleton<IDashboardEndpointProvider, HostDashboardEndpointProvider>();
```

Comment:
`HostDashboradEndpointProvider` is being added regardless of whether the dashboard is enabled.